executors/sshexec/block_volume.go

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright (c) 2017 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package sshexec
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/heketi/heketi/executors"
+	"github.com/lpabon/godbc"
+)
+
+func (s *SshExecutor) BlockVolumeCreate(host string,
+	volume *executors.BlockVolumeRequest) (*executors.BlockVolumeInfo, error) {
+
+	godbc.Require(volume != nil)
+	godbc.Require(host != "")
+	godbc.Require(volume.Name != "")
+
+	type CliOutput struct {
+		Iqn      string   `json:"IQN"`
+		Username string   `json:"USERNAME"`
+		Password string   `json:"PASSWORD"`
+		Portal   []string `json:"PORTAL(S)"`
+		Result   string   `json:"RESULT"`
+		ErrCode  int      `json:"errCode"`
+		ErrMsg   string   `json:"errMsg"`
+	}
+
+	var auth_set string
+	if volume.Auth {
+		auth_set = "enable"
+	} else {
+		auth_set = "disable"
+	}
+
+	cmd := fmt.Sprintf("gluster-block create %v/%v  ha %v auth %v  %v %vG --json",
```

Comment:
@MohamedAshiqrh I think, gluster-block expects `GiB` instead of "G".

Reply:
We need to call gluster-block with "prealloc full"